Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_baking.py

```diff
@@ -0,0 +1,1202 @@
+        return getattr(obj, attr, *args)
+
+    return functools.reduce(_getattr, [obj] + attr.split('.'))
+
+
+# Setting classes:
+class MESHSYNC_BakeChannelSetting(bpy.types.PropertyGroup):
+    name: bpy.props.StringProperty(name="")
+    bakeChannelEnabled: bpy.props.BoolProperty(name="", description="Whether to bake this channel or not", default=True)
+
+
+def msb_bakeAllGet(meshsync_bake_settings):
+    for channelSetting in meshsync_bake_settings.bake_channel_settings:
+        if not channelSetting.bakeChannelEnabled:
+            return False
+    return True
+
+
+def msb_bakeAllSet(meshsync_bake_settings, newValue):
+    for channelSetting in meshsync_bake_settings.bake_channel_settings:
+        channelSetting.bakeChannelEnabled = newValue
+
+
+class MESHSYNC_BakeSettings(bpy.types.PropertyGroup):
+    '''
+    Groups all bake settings in a single class.
+    '''
+    bakedTexturesPath: bpy.props.StringProperty(name="Baked texture path", default='')
+    bakedTextureSize: bpy.props.IntVectorProperty(name="Baked texture size", size=2,
+                                                  default=(512, 512))
+    bake_selection: bpy.props.EnumProperty(name="Objects to bake",
+                                           items=(('ALL', 'All',
+                                                   'Bake all objects in the scene (including hidden)'),
+                                                  ('SELECTED', 'Selected',
+                                                   'Bake only the selected object')),
+                                           default='ALL')
+    bake_channel_settings: bpy.props.CollectionProperty(type=MESHSYNC_BakeChannelSetting)
+    bake_all_channels: bpy.props.BoolProperty(name="All", description="Toggle all",
+                                              get=msb_bakeAllGet,
+                                              set=msb_bakeAllSet)
+    generate_uvs: bpy.props.EnumProperty(name="Generate UVs",
+                                         description="Whether to auto-generate UVs",
+                                         items=(('OFF', 'Off',
+                                                 'Bake all objects in the scene (including hidden)'),
+                                                ('IF_NEEDED', 'If needed',
+                                                 'Automatically UV unwraps objects if there are no UVs or existing UVs are not in the 0..1 range. WARNING: This will delete existing UVs on the object!'),
+                                                ('ALWAYS', 'Always',
+                                                 'Always automatically UV unwraps objects. WARNING: This will delete existing UVs on the object!')),
+                                         default='OFF')
+    apply_modifiers: bpy.props.BoolProperty(name="Apply modifiers",
+                                            description="In order to bake and get correct UVs, all modifiers need to be applied",
```

Comment:
Should we add a warning here as well, similar to the one about UVs?

Reply:
Yeah, done